### DISCUSSION TITLE
etcd instllation in Debian

### DISCUSSION THREAD

**Discussion by skizilbash** [etcd-io/etcd#17670] [Q&A] [answered]
### Bug report criteria

- [X] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [ ] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [ ] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [ ] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

Hello,

I am trying to install etcd for patroni in Debium 6.1.0-17-amd64. I get the below error. 
After reading quite a few blogs/forums seems like many people reported that this was a bug in previous version but fixed in 3.5.0. Apparently that's not the case hence i am reporting it. Please share a possible fix for this.

root@patroni-pgcluster-01:/var/rpms/etcd# ./build.sh
./scripts/test_lib.sh: line 5: go: command not found
must be run from 'go.etcd.io/etcd/v3' module directory

Thanks.
Shawn K

Below is my go.mod and as you can see i've modified it by adding but it didn't fix the issue.

"replace google.golang.org/grpc => google.golang.org/grpc v1.26.0"

root@patroni-pgcluster-01:/var/rpms/etcd# cat go.mod
module go.etcd.io/etcd/v3

go 1.16

replace (
        go.etcd.io/etcd/api/v3 => ./api
        go.etcd.io/etcd/client/pkg/v3 => ./client/pkg
        go.etcd.io/etcd/client/v2 => ./client/v2
        go.etcd.io/etcd/client/v3 => ./client/v3
        go.etcd.io/etcd/etcdctl/v3 => ./etcdctl
        go.etcd.io/etcd/etcdutl/v3 => ./etcdutl
        go.etcd.io/etcd/pkg/v3 => ./pkg
        go.etcd.io/etcd/raft/v3 => ./raft
        go.etcd.io/etcd/server/v3 => ./server
        go.etcd.io/etcd/tests/v3 => ./tests
)

replace google.golang.org/grpc => google.golang.org/grpc v1.26.0

require (
        github.com/bgentry/speakeasy v0.1.0
        github.com/dustin/go-humanize v1.0.0
        github.com/spf13/cobra v1.1.3
        go.etcd.io/bbolt v1.3.6
        go.etcd.io/etcd/api/v3 v3.5.0
        go.etcd.io/etcd/client/pkg/v3 v3.5.0
        go.etcd.io/etcd/client/v2 v2.305.0
        go.etcd.io/etcd/client/v3 v3.5.0
        go.etcd.io/etcd/etcdctl/v3 v3.5.0
        go.etcd.io/etcd/etcdutl/v3 v3.5.0
        go.etcd.io/etcd/pkg/v3 v3.5.0
        go.etcd.io/etcd/raft/v3 v3.5.0
        go.etcd.io/etcd/server/v3 v3.5.0
        go.etcd.io/etcd/tests/v3 v3.5.0
        go.uber.org/zap v1.17.0
        golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
        google.golang.org/grpc v1.38.0
        gopkg.in/cheggaaa/pb.v1 v1.0.28
)

### What did you expect to happen?

i expected the ./build.sh script to run without errors.

### How can we reproduce it (as minimally and precisely as possible)?

run ./build.sh after installing etcd binaries

### Anything else we need to know?

n/a

### Etcd version (please run commands below)

3.5.12

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

### Relevant log output

```Shell
root@patroni-pgcluster-01:/var/rpms/etcd# ./build.sh
./scripts/test_lib.sh: line 5: go: command not found
must be run from 'go.etcd.io/etcd/v3' module directory
```

**Comment by skizilbash** [comment]
root@patroni-pgcluster-01:/var/rpms/etcd# which go
/usr/bin/go

root@patroni-pgcluster-01:/var/rpms/etcd# ./build.sh
go: errors parsing go.mod:
/var/rpms/etcd/go.mod:41:9: usage: require module/path v1.2.3
must be run from 'go.etcd.io/etcd/v3' module directory

**Comment by ivanvc** [comment, marked as answer]
Hi @skizilbash,

If you're considering installing the v3.5 release, I suggest going with the latest one, v3.5.12. Also, if you're not looking into developing, using the pre-compiled binaries is best. The website has instructions on how to do this: https://etcd.io/docs/v3.5/install/#install-pre-built-binaries.

I think a better place for this question would be the [discussion forums](https://github.com/etcd-io/etcd/discussions).

**Reply by jmhbnz** [reply]
+1 - Please use the latest 3.5.x release and let us know if you are still having issues.

**Comment by skizilbash** [comment]
solved
